### PR TITLE
Use http when relaying selenium commands through sauce connect tunnel

### PR DESCRIPTION
**Pull request by ndmanvar**

### CODE REVIEW

**Comment by ndmanvar** [comment]
We should be using `http` when selenium commands are being relayed through sauce connect tunnel (i.e. when `this.config_.sauceSeleniumAddress` is specified).
Use `https://SAUCE_USERNAME:SAUCE_ACCESS_KEYondemand.saucelabs.com:443/wd/hub` when `this.config_.sauceSeleniumAddress` is NOT specified.

**Comment by ndmanvar** [comment]
I signed it!

**Comment by ndmanvar** [comment]
I signed it!

**Comment by ndmanvar** [comment]
@mgiambalvo Can you please review and merge? Happy to clarify and answer any questions as needed (:

Hope all is well with you!

**Comment by ndmanvar** [comment]
@wyvern8 I don't think that is necessary, as people were relaying through sauce connect tunnel due to lack of having 'https' endpoint. Now since we have the 'https' endpoint, users have have moved to using 'https://ondemand.saucelabs.com' and using their internal proxy if needed.

Sauce wants to discourage the use of relaying through tunnels (for various reasons) and have them go to 'https://ondemands.saucelabs.com' and use company/corporate proxy to get there if needed.

Let me know if you that makes sense or if you have any questions. Also, it might be likely that I am missing something or misunderstanding (:.

With the current code in place, relaying through sauce connect tunnel would not work as it will try to go through 'https://USERNAME:ACCESS_KEY@sauceconnect.tunnel.com:SELENIUM_LISTENER_PORT' and SELENIUM_LISTENER_PORT will not be 443.

**Comment by wyvern8** [comment]
OK. Makes sense.  Agree we should get this in now, and can look at options later if people need it.

**Comment by ndmanvar** [comment]
@wyvern8 Sounds good. What are the next steps for getting this in? Can you merge this in?

File: lib/driverProviders/sauce.ts [hunk 1]
```diff
@@ -63,10 +63,10 @@ export class Sauce extends DriverProvider {
     this.config_.capabilities['username'] = this.config_.sauceUser;
     this.config_.capabilities['accessKey'] = this.config_.sauceKey;
     this.config_.capabilities['build'] = this.config_.sauceBuild;
-    let auth = 'https://' + this.config_.sauceUser + ':' + this.config_.sauceKey + '@';
-    this.config_.seleniumAddress =
-        auth + (this.config_.sauceSeleniumAddress ? this.config_.sauceSeleniumAddress :
-                                                    'ondemand.saucelabs.com:443/wd/hub');
+    let auth = this.config_.sauceUser + ':' + this.config_.sauceKey + '@';
+    this.config_.seleniumAddress = this.config_.sauceSeleniumAddress ?
+        ('http://' + auth + this.config_.sauceSeleniumAddress) :
```
Comment (cnishina) [inline lib/driverProviders/sauce.ts]: The `(` and `)` are not necessary in both the ternary 2nd and 3rd argument and should be removed.

Comment (wyvern8) [inline lib/driverProviders/sauce.ts]: @ndmanvar i just noticed that we also need this fix to support protractor 5.1.1+ as the new forcing to https for ondemand does not work through our inspection proxy.   We would need to set sauceSeleniumAddress to ondemand.saucelabs.com:80/wd/hub to work around this.. (unless we make the scheme another parameter as mentioned previously..)  - if you can at least make the style fix suggested by @cnishina that would be great.

**Comment by ndmanvar** [comment]
Fixed and rebased @wyvern8 and @wyvern8 !

Look forward to issuing more PRs to Protractor.
Keep up the great work (:

**Comment by ndmanvar** [comment]
Circle CI build is failing due to jasmine timeouts / test taking too long (unrelated to this change).

```
...............................[04:37:09] I/launcher - Running 1 instances of WebDriver
[04:37:09] I/hosted - Using the selenium server at http://localhost:4444/wd/hub
Started
FA Jasmine spec timed out. Resetting the WebDriver Control Flow.


Failures:
1) timeout spec should timeout due to jasmine spec limit
  Message:
    Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
  Stack:
    Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
        at ontimeout (timers.js:365:14)
        at tryOnTimeout (timers.js:237:5)
        at Timer.listOnTimeout (timers.js:207:5)

1 spec, 1 failure
Finished in 1.415 seconds

[04:37:11] I/launcher - 0 instance(s) of WebDriver still running
[04:37:11] I/launcher - chrome #01 failed 1 test(s)
[04:37:11] I/launcher - overall: 1 failed spec(s)
[04:37:11] E/launcher - Process exited with error code 1

>>> fail: Error: expecting test max duration: 1000, actual: 1402
```

**Comment by wyvern8** [comment]
@cnishina how should we progress this?  Do you have other suggestions, or should @ndmanvar rebase and try again?  If we want to look at setting protocol/scheme as a separate parameter that is doable too.

**Comment by ndmanvar** [comment]
@wyvern8 @cnishina Just merged master back in and all checks have passed.

**Comment by mgiambalvo** [comment]
Sorry for the extreme delay on this. Could you please rebase and push, and then I'll merge it in. Thanks for your help!

**Comment by ndmanvar** [comment]
I signed it!

**Comment by ndmanvar** [comment]
I signed it

**Comment by ndmanvar** [comment]
Closing this and opening a new PR, as I am having Google CLA email / commit problems. @heathkit

**Comment by ndmanvar** [comment]
Closing this one and re-opening as https://github.com/angular/protractor/pull/4228 in order to solve Google CLA email / commit alignment problems (I changed the email and recommitted, and issued a new PR).

**Pull request closed without merging**